desertbot/modules/commands/OutOfContext.py

```diff
@@ -0,0 +1,244 @@
+        return self._getQuote(message.replyTo, "", False, -1)
+
+    def _removeQuote(self, source, quote):
+        self.storage[source].remove(quote)
+        return IRCResponse(ResponseType.Say, f"Quote '{quote}' was removed from the log!", source)
+
+    def _postList(self, source, searchString, searchNickname):
+        if len(self.storage) == 0 or source not in self.storage:
+            return IRCResponse(ResponseType.Say, "There are no quotes in the log.", source)
+        regex = re2.compile(searchString, re2.IGNORECASE)
+        matches = []
+        if searchNickname:
+            for x in self.storage[source]:
+                if x[21] == "*":
+                    match = re2.search(regex, x[:x.find(" ", 23)])
+                else:
+                    match = re2.search(regex, x[x.find("<") + 1:x.find(">")])
+                if match:
+                    matches.append(x)
+        else:
+            for x in self.storage[source]:
+                if re2.search(regex, x[x.find(">") + 1:]):
+                    matches.append(x)
+        if len(matches) == 0:
+            return IRCResponse(ResponseType.Say, f"No matches for '{searchString}' found.", source)
+
+        pasteLink =  self.bot.moduleHandler.runActionUntilValue('upload-dbco',
+                                                                string.stripFormatting("\n".join(matches)),
+                                                                10 * 60)
+
+        return IRCResponse(ResponseType.Say, f"Link posted! (Expires in 10 minutes) {pasteLink}.", source)
+
+    def _getQuote(self, source, searchString, searchNickname, index):
+        if len(self.storage) == 0 or source not in self.storage:
+            return IRCResponse(ResponseType.Say, "There are no quotes in the log.", source)
+        regex = re2.compile(searchString, re2.IGNORECASE)
+        matches = []
+        if searchNickname:
+            for x in self.storage[source]:
+                if x[21] == "*":
+                    match = re2.search(regex, x[:x.find(" ", 23)])
+                else:
+                    match = re2.search(regex, x[x.find("<") + 1:x.find(">")])
+                if match:
+                    matches.append(x)
+        else:
+            for x in self.storage[source]:
+                if re2.search(regex, x[x.find(">") + 1:]):
+                    matches.append(x)
+        if len(matches) == 0:
+            return IRCResponse(ResponseType.Say, f"No matches for '{searchString}' found.", source)
+        if index < 0 or index > len(matches) - 1:
+            index = random.randint(0, len(matches) - 1)
+        return IRCResponse(ResponseType.Say, f"Quote #{index + 1}/{len(matches)}: {matches[index]}", source)
+
+    def _unrecognizedSubcommand(self, subCommand):
+        return (f"unrecognized subcommand f'{subCommand}', "
+                f"available subcommands for alias are: {', '.join(self.subCommands.keys())}")
+
+    subCommands = OrderedDict([
```

Comment:
Might be a good idea to have the subCommands dict at the bottom of the file, so it's easy to spot? I know python doesn't like it if you try to have it at the top of the file for some reason...